docs/tutorial/bigquery.md

```diff
@@ -0,0 +1,153 @@
+
+.. thumbnail:: ../media/databases/big-query-1.png
+   :width: 32%
+
+.. thumbnail:: ../media/databases/big-query-2.png
+   :width: 32%
+
+.. thumbnail:: ../media/databases/big-query-3.png
+   :width: 32%
+```
+
+## Create a service account & key file
+
+To use the BigQuery API from the Streamlit Cloud, you need a Google Cloud Platform service account (a special account type for programmatic data access). Go to the [Service Accounts](https://console.cloud.google.com/iam-admin/serviceaccounts) page and create an account with the **Viewer** permission (this will let the account access data but not change it):
+
+```eval_rst
+.. thumbnail:: ../media/databases/big-query-4.png
+   :width: 32%
+
+.. thumbnail:: ../media/databases/big-query-5.png
+   :width: 32%
+
+.. thumbnail:: ../media/databases/big-query-6.png
+   :width: 32%
+
+.. note:: If the button **CREATE SERVICE ACCOUNT** is gray, you don't have the correct permissions. Ask the admin of your Google Cloud project for help.
+```
+
+After clicking **DONE**, you should be back on the service accounts overview. Create a JSON key file for the new account and download it:
+
+```eval_rst
+.. thumbnail:: ../media/databases/big-query-7.png
+   :width: 32%
+
+.. thumbnail:: ../media/databases/big-query-8.png
+   :width: 32%
+
+.. thumbnail:: ../media/databases/big-query-9.png
+   :width: 32%
+```
+
+```eval_rst
+.. important:: Store the key file in a safe location and don't share it with anyone.
+```
+
+## Add the key file to your local app secrets
+
+Your local Streamlit app will read secrets from a file `.streamlit/secrets.toml` in your app's root directory. Create this file if it doesn't exist yet and add the content of the key file you just downloaded to it as shown below:
+
+```
+# .streamlit/secrets.toml
+
+[gcp_service_account]
+type = "service_account"
+project_id = "xxx"
+private_key_id = "xxx"
+private_key = "xxx"
+client_email = "xxx"
+client_id = "xxx"
+auth_uri = "https://accounts.google.com/o/oauth2/auth"
+token_uri = "https://oauth2.googleapis.com/token"
+auth_provider_x509_cert_url = "https://www.googleapis.com/oauth2/v1/certs"
+client_x509_cert_url = "xxx"
+```
+
+```eval_rst
+.. important:: Add this file to .gitignore and don't commit it to your Github repo!
+```
+
+## Copy your app secrets to the cloud
+
+As the `secrets.toml` file above is not committed to Github, you need to pass its content to your deployed app (on Streamlit sharing or Streamlit for Teams) separately. Go to the [app dashboard](https://share.streamlit.io/) and in the app's dropdown menu, click on **Edit Secrets**. Copy the content of `secrets.toml` into the text area. More info in [Secrets Management](../deploy_streamlit_app.html#secrets-management).
+
+![](../media/databases/big-query-10.png)
+
+## Install [google-cloud-bigquery](https://googleapis.dev/python/bigquery/latest/index.html) and add it to your requirements file
+
+```
+pip install google-cloud-bigquery
```

Comment:
Just to clarify: do users following the tutorial not have to pip install the module from their terminal? 

As the tutorial concerns deploying the app on S4A or S4T, adding `google-cloud-bigquery` to `requirements.txt` is sufficient, right? If so, we can remove the `pip install` line and change the heading and content to:

## Add google-cloud-bigquery to your requirements file

Add the [google-cloud-bigquery](https://googleapis.dev/python/bigquery/latest/index.html) package to your `requirements.txt` file, preferably pinning its version (replace `x.x.x` with the version want installed):

```
# requirements.txt
google-cloud-bigquery==x.x.x
```

Reply:
They do need to install it locally, just that with the code formatting the way it is, it wasn't clear whether that code was to be run in a terminal vs. add it to requirements.txt. From the header, it would seem like you are supposed to add it to requirements.txt.

I agree that since the tutorials are focused for s4a/t, then we can probably remove the pip piece altogether